python-avd/pyavd/api/pool_manager/base_classes.py

```diff
@@ -0,0 +1,295 @@
+            )
+
+        # currently we only support int pools
+        return False
+
+    def _next_available(self) -> T_ValueType:
+        if self.collection.value_type is int:
+            collection = cast(PoolCollection[T_PoolKeyType, T_AssignmentKeyType, int], self.collection)
+            assignments = cast(dict[T_AssignmentKeyType, PoolAssignment[T_AssignmentKeyType, int]], self.assignments)
+            existing_ids = {assignment.value for assignment in assignments.values()}
+            # Create a filterfalse generator from a range starting from the min_value, excluding the values that are already assigned.
+            # Nothing will be iterated at this point, but the next(iter()) below will ask the generator for the first item.
+            available_ids = filterfalse(existing_ids.__contains__, range(collection.min_value, collection.min_value + len(existing_ids) + 2))
+            next_available = next(iter(available_ids))
+            if collection.max_value is not None and next_available > collection.max_value:
+                msg = (
+                    f"No available values found within '{self.collection.pools_key}' with key '{self.pool_key}' "
+                    f"and range {collection.min_value}-{collection.max_value}."
+                )
+                raise ValueError(msg)
+            return cast(T_ValueType, next_available)
+
+        msg = f"Currently we only support pools of Integers. Got {self.collection.value_type}."
+        raise NotImplementedError(msg)
+
+    def get_assignment(self, key: T_AssignmentKeyType, requested_value: T_ValueType | None = None) -> PoolAssignment[T_AssignmentKeyType, T_ValueType]:
+        """
+        Returns the assignment for the key if found in the pool.
+
+        Otherwise a new entry is inserted into the pool and returned.
+
+        If 'requested_value' is given and available in the pool, any existing assignment will be updated and returned.
+        For new assignments this value will be used.
+        """
+        if key not in self.assignments:
+            # No existing assignment. Use the requested value if given and available. Otherwise use next available.
+            new_assignment_value = requested_value if requested_value is not None and self._is_value_available(requested_value) else self._next_available()
+            self.assignments[key] = self.collection.assignment_cls(key=key, value=new_assignment_value, accessed=True)
+            self.collection.changed = True
+        elif requested_value is not None and requested_value != self.assignments[key].value and self._is_value_available(requested_value):
+            # Existing assignment but not with the requested value. Change the existing assignment to the requested_value.
+            self.assignments[key] = self.collection.assignment_cls(key=key, value=requested_value, accessed=True)
+            self.collection.changed = True
```

Comment:
would be cool to be able to have the previous call setting the bool `self.collection.changed`

Maybe a help funciton 
```
self.collection.assign(key, value):

class Collection
  def assign(self, key, value):
    self.assignment_cls(key=key, value=new_assignment_value, accessed=True)
    self.changed = True
```

Reply:
awe will look more into it later - at least when we add some other classes.